EvaluteRTE: Correctly report AbortOnValue aborts with negative abort codes

t-b opened the pull request:
We need to assume that all error codes not being -1, -2, -3, -4 are from AbortOnValue as we can't differentiate them. This now correctly reports AbortOnValue aborts with negative abort codes.

Broken since a7ab54c (More detailed RTE error message, 2017-03-07). It did not help that the Igor Pro documentation is also wrong in this regard.

Garados007 commented on procedures/igortest-basics.ipf:
```diff
@@ -453,15 +453,14 @@ static Function EvaluateRTE(err, errmessage, abortCode, funcName, funcType, proc
 				IUTF_Reporting#AddError(str, IUTF_STATUS_ERROR)
 				break
 			default:
+				if(abortCode != 0)
```
This can be optimized. Just add a case which checks for 0 and does nothing and move the content of the if statement into the default case directly.

Doing this you have one level of indentation less and no if statement.

t-b replied:
Good point. But thinking about this again, the dheck for non-zero abort code is not correct either. The abort code can be anything. Fill send a fixed version.

MichaelHuth replied:
Ok, so the -1, -2, -3 cases can possibly also be AbortOnValue then.

t-b replied:
Unfortunately yes. My idea here is that this is not as likely as the other abort types.

t-b replied:
Should be fixed.

t-b committed:
EvaluteRTE: Correctly report AbortOnValue aborts with negative abort codes

We need to assume that all error codes not being -1, -2, -3, -4 are from
AbortOnValue as we can't differentiate them. This now correctly reports
AbortOnValue aborts with negative abort codes.

Broken since a7ab54c (More detailed RTE error message, 2017-03-07). It did
not help that the Igor Pro documentation is also wrong in this regard.

t-b committed:
EvaluteRTE: Correctly report AbortOnValue aborts with negative abort codes

We need to assume that all error codes not being -1, -2, -3, -4 are from
AbortOnValue as we can't differentiate them. This now correctly reports
AbortOnValue aborts with negative abort codes.

Broken since a7ab54c (More detailed RTE error message, 2017-03-07). It did
not help that the Igor Pro documentation is also wrong in this regard.

t-b committed:
TAP_AreAllFunctionsSkip: Remove invalid parameter documentation

The parameter is gone since 2f80d1a (Refactor TAP to use test result
waves, 2022-12-02).

t-b committed:
TAP_AreAllFunctionsSkip: Remove invalid parameter documentation

The parameter is gone since 2f80d1a (Refactor TAP to use test result
waves, 2022-12-02).

t-b committed:
.pre-commit-config.yaml: Update reuse-tool

The old version does not work with the updated docker-utils CI image.

$ pre-commit run --all-files --show-diff-on-failure
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[WARNING] repo `https://github.com/pre-commit/pre-commit-hooks` uses deprecated stage names (commit, push) which will be removed in a future version.  Hint: often `pre-commit autoupdate --repo https://github.com/pre-commit/pre-commit-hooks` will fix this.  if it does not -- consider reporting an issue to that repo.
[INFO] Initializing environment for https://github.com/sirosen/texthooks.
[INFO] Initializing environment for https://github.com/fsfe/reuse-tool.
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/sirosen/texthooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/fsfe/reuse-tool.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
check for added large files..............................................Passed
fix utf-8 byte order marker..............................................Passed
check for case conflicts.................................................Passed
check that executables have shebangs.....................................Passed
check json...........................................(no files to check)Skipped
check for merge conflicts................................................Passed
check that scripts with shebangs are executable..........................Passed
check for broken symlinks............................(no files to check)Skipped
check yaml...............................................................Passed
detect destroyed symlinks................................................Passed
fix end of files.........................................................Passed
mixed line ending........................................................Passed
trim trailing whitespace.................................................Passed
Forbid the use of unicode BiDi control characters........................Passed
reuse....................................................................Failed
- hook id: reuse
- exit code: 1
Traceback (most recent call last):
  File "/builds/mirror/igortest/.cache/pre-commit/repoaq9hl26l/py_env-python3/bin/reuse", line 5, in <module>
    from reuse._main import main
  File "/builds/mirror/igortest/.cache/pre-commit/repoaq9hl26l/py_env-python3/lib/python3.13/site-packages/reuse/__init__.py", line 15, in <module>
    from pkg_resources import DistributionNotFound, get_distribution
ModuleNotFoundError: No module named 'pkg_resources'
